src/main/java/com/mraof/minestuck/alchemy/PredeterminedCaptchaLoader.java

```diff
@@ -0,0 +1,96 @@
+package com.mraof.minestuck.alchemy;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.gson.*;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.JsonOps;
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimplePreparableReloadListener;
+import net.minecraft.util.profiling.ProfilerFiller;
+import net.minecraft.world.item.Item;
+import net.minecraftforge.event.AddReloadListenerEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.*;
+
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mod.EventBusSubscriber(modid = Minestuck.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE)
+public final class PredeterminedCaptchaLoader extends SimplePreparableReloadListener<Map<String, ResourceLocation>>
+{
+	private static final Logger LOGGER = LogManager.getLogger();
+	public static final String PATH = "minestuck/captcha_codes.json";
+	
+	@Override
+	protected Map<String, ResourceLocation> prepare(ResourceManager resourceManager, ProfilerFiller profiler)
+	{
+		Map<String, ResourceLocation> captchaData = new HashMap<>();
+		
+		for(String namespace : resourceManager.getNamespaces())
+		{
+			parseCaptchaCodesFromNamespace(resourceManager, namespace).ifPresent(parsedData -> {
+				for(String captcha : parsedData.keySet())
+					captchaData.put(captcha, parsedData.get(captcha));
```

Comment:
Overlap in captchas and items between namespaces could be troublesome. Captchas and items need to be checked against the predefined ones that has already been added to solve that.